backend/gateway/meta/meta.go

```diff
@@ -119,6 +119,77 @@ func ResourceNames(pb proto.Message) []*auditv1.Resource {
 	return nil
 }
 
+// ExtractProtoPatternsValues takes a proto and returns its pattern populated with values
+func ExtractProtoPatternsValues(pb proto.Message) string {
+	m := pb.ProtoReflect()
+	opts := m.Descriptor().Options().ProtoReflect()
+
+	populatedPattern := []string{}
+
+	if opts.Has(identifierTypeDescriptor) {
```

Comment:
can/should we share any code with `func ResourceNames`?

Reply:
Yeah let me see what we can share